assets/app/scripts/directives/oscFileInput.js

```diff
@@ -10,15 +10,121 @@ angular.module('openshiftConsole')
         disabled: "=ngDisabled",
         // Show the file contents below the file input.
         showValues: "=",
-        helpText: "@?"
+        helpText: "@?",
+        dropZoneId: "@?",
+        dragging: "=",
       },
       templateUrl: 'views/directives/osc-file-input.html',
       link: function(scope, element){
         scope.helpID = _.uniqueId('help-');
         scope.supportsFileUpload = (window.File && window.FileReader && window.FileList && window.Blob);
         scope.uploadError = false;
-        $(element).change(function(){
-          var file = $('input[type=file]', this)[0].files[0];
+
+        var droppableElement = (scope.dropZoneId) ? $('#' + scope.dropZoneId) : element,
+        dropZoneName = scope.dropZoneId + "-drag-and-drop-zone",
+        dropZoneSelectorName = "#" + dropZoneName,
+        highlightDropZone = false,
+        showDropZone = false,
+        addDropZone = false;
+
+        scope.$watch('disabled', function() {
```

Comment:
I would even consider adding it once globally... would be nice to react with a "file upload not enabled on this page" message if they drag in a file when we're not expecting it